torch/csrc/autograd/engine.cpp

```diff
@@ -364,6 +278,9 @@ auto Engine::thread_main(GraphTask *graph_task) -> void {
       try {
         evaluate_function(task);
       } catch (std::exception& e) {
+        LOG(WARNING)
+            << "Autograd engine received exception while executing func "
+            << task.fn_->name() << ": " << e.what();
```

Comment:
I suspect this affects the error reporting for Python users of autograd. Have you tried? Just make a custom function that raises an exception.

Reply:
My general feeling is that if you continue to propagate an exception, you shouldn't also log a warning for it (whether or not to log is a decision that should be made at the site that actually handles the exception in the end.)

(I also think we should use `TORCH_WARN` here, that way the warnings show up properly as Python warnings and can be handled using Python warning mechanisms.)

Reply:
(More on parenthetical at https://github.com/pytorch/pytorch/issues/27608 )

Reply:
I see, I'll remove the logging from here for now since we propagate the exception.